packages/modules/promotion/src/services/promotion-module.ts

```diff
@@ -522,6 +526,37 @@ export default class PromotionModuleService<
         )
       }
 
+      if (!campaignData && !campaignId) {
+        promotionsData.push({ ...promotionData })
+
+        continue
+      }
+
+      const existingCampaign = existingCampaigns.find(
+        (c) => c.id === campaignId
+      )
+
+      if (campaignId && !existingCampaign) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `Could not find campaign with id - ${campaignId}`
+        )
+      }
+
+      const campaignCurrency =
+        campaignData?.currency ||
+        existingCampaigns.find((c) => c.id === campaignId)?.currency
+
+      if (
+        !campaignCurrency ||
+        campaignCurrency !== applicationMethodData?.currency_code
+      ) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Currency between promotion and campaigns should match`
+        )
```

Comment:
Goes along with my question on the data model. If the campaign is of type spend, do we need to enforce this match.